### PR TITLE
OOP school library: refactor your code

### PR DESCRIPTION
**Pull request by rodrigopenayo98**
- Refinement of the codebase to enhance maintainability and readability.
- Adherence to software development best practices, including YAGNI (You Ain't Gonna Need It), KISS (Keep It Simple, Stupid), DRY (Don't Repeat Yourself), and the Ruby style guide.
- Application of SOLID principles, with a special focus on the Single-responsibility Principle.
- The primary focus is on streamlining the main.rb file, ensuring its main function is to serve as an entry point, prompting user choices, and initiating the application.
- Relevant pieces of logic have been thoughtfully extracted into separate classes.
- This process improves code clarity, fosters modular development, and sets the stage for easier future enhancements and maintenance.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -148,6 +148,22 @@ To run linter tests, run the following command:
 - Twitter: [@rodrigopenayo98](https://twitter.com/rodrigopenayo98)
 - LinkedIn: [LinkedIn](https://www.linkedin.com/in/rodrigo-penayo-391226158/)
 
+
```
Comment (HAMMAS-SALEEM) [inline README.md:151]: - [ ] Great work so far 👍 but i guess you didn't understand the requirements well 🤔 It is a pair-programming project means every member of this team should commit code from his/her side into the codebase but i see that only one member pushed code but other two members only fix the linters and readme file but didn't push any code from their side 😨 Kindly all members push some code into this pull request to fulfill the pair-programming requirement 👍

**Review by HAMMAS-SALEEM — COMMENTED** [review]
# Invalid Submission :red_circle:

invalid-submission

## Hey Team (@rodrigopenayo98, @kit0-0, @leloufadel) :wave:

Good job on implementing some of the requirements but I cannot proceed review of your pull request because there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (rodrigopenayo98) [reply README.md:151]: hello @HAMMAS-SALEEM ! how are you? Look, we all collaborated... there wasn't much to do, the concept of pair-programming does not necessarily indicate that everyone must add code, but rather that they collaborate on the same project.
Anyway, the girls will add code in the following group projects! we will rotate. We work this way on many occasions and it is the first time we receive that type of correction.

**Review by ShadyShawkat — APPROVED** [review]
## Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**